Sources/MongoSwift/MongoCollection+Indexes.swift

```diff
@@ -406,16 +412,14 @@ extension MongoCollection {
      *    - `LogicError` if the provided session is inactive.
      *    - `LogicError` if this collection's parent client has already been closed.
      */
-    public func listIndexNames(session _: ClientSession? = nil) -> EventLoopFuture<[String]> {
-        return self.listIndexes().flatMap { cursor in
-            cursor.toArray()
-        }.flatMapThrowing { models in
-            try models.map { model in
-                guard let name = model.options?.name else {
+    public func listIndexNames(session: ClientSession? = nil) -> EventLoopFuture<[String]> {
+        let operation = ListIndexesOperation(collection: self, nameOnly: true)
+        return self._client.operationExecutor.execute(operation, client: self._client, session: session)
+            .flatMapThrowing { result in
+                guard case let .names(names) = result else {
                     throw InternalError(message: "Server response missing a 'name' field")
```

Comment:
I think this error case now corresponds to us getting back an unexpected result type (i.e. `.models` rather than `.names`), so we could update the error message here to match the "invalid result" one you used above.